File: frame/map-frame-project/MapFrameCoreModule/src/main/java/com/grandtech/mapframe/core/GMap.java
package com.grandtech.mapframe.core;

import android.content.Context;
import android.util.Log;

import com.grandtech.mapframe.core.rules.Rules;
import com.mapbox.mapboxsdk.Mapbox;

/**
 * @ClassName GMap
 * @Description TODO 初始化mapbox实例
 * @Author: fs
 * @Date: 2021/2/19 14:01
 * @Version 2.0
 */
public class GMap implements Rules {
    /**
     * TODO 获取实例
     * @param context
     */
    public static void getInstance(Context context){
        Mapbox.getInstance(context, context.getString(R.string.mapbox_access_token));
    }
}

File: frame/map-frame-project/MapFrameCoreModuleV10/src/main/java/com/gradtech/mapframev10/core/GMap.kt
package com.gradtech.mapframev10.core

import android.content.Context
import com.gradtech.mapframev10.R
import com.mapbox.maps.ResourceOptionsManager
import com.mapbox.maps.TileStoreUsageMode

/**
 * @ClassName GMap
 * @Description TODO
 * @Author: fs
 * @Date: 2022/8/2 10:25
 * @Version 2.0
 */
object GMap {

    @JvmStatic
    fun getInstance( context : Context){
        ResourceOptionsManager.getDefault(context, context.getResources().getString(R.string.mapbox_access_token)).update {
            tileStoreUsageMode(TileStoreUsageMode.READ_ONLY)
        }
    }
}